### PR TITLE
Add reset for vector

### LINKED ISSUE #51: [FEATURE] Provide possibility to unset value for vector field(reset)
**Is your feature request related to a problem?**
https://github.com/opendistro-for-elasticsearch/k-NN/issues/343

**What solution would you like?**
We would like to provide ability to accept null values to reset/remove the vector field value

### PR DESCRIPTION
**Pull request by martin-gaievski**
Add ability to reset vector value by using 'null'

### Description
Adding possibility to unset vector's value by updating vector field with 'null' value
 
### Issues Resolved
#51 
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java [hunk 1]
```diff
@@ -399,4 +399,46 @@ public void testExistsQuery() throws IOException {
...
     }
+
+    public void testIndexingVectorValidation_updateVectorWithNull() throws Exception {
+        Settings settings = Settings.builder()
+                .put(getKNNDefaultIndexSettings())
+                .build();
+
+        createKnnIndex(INDEX_NAME, settings, createKnnIndexMapping(FIELD_NAME, 4));
+
+        // valid case with 4 dimension
```
Comment (VijayanB) [inline src/test/java/org/opensearch/knn/index/OpenSearchIT.java:410]: shall we add test case with two documents and k =1, where after update, you will get other document as search response?

Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java:410]: I think such test case makes sense, I can extend existing test case and start with two documents instead of one, then after update search will return the other document that wasn't updated. Part with retrieving updated document by id should remains the same

Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java:410]: addressed your comment in next commit - https://github.com/opensearch-project/k-NN/pull/237/commits/a038333b8ac8d982a383234336b8ac7dea5b02c3, please check

**Review by VijayanB — APPROVED** [review]

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java (hunk 1 above)
Comment (VijayanB) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java:410]: Thank you!

**Comment by martin-gaievski** [comment]
Added a bit more tests after discussing with Jack offline, scenario added: after vector has been reset to null we assign some value to vector again. After that document should be discoverable by search and when doc is retrieved the vector value should be updated

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java [hunk 2]
```diff
@@ -399,4 +399,71 @@ public void testExistsQuery() throws IOException {
...
+        final Response response = searchKNNIndex(INDEX_NAME, knnQueryBuilder, k);
+        final List<KNNResult> results = parseSearchResponse(EntityUtils.toString(response.getEntity()), FIELD_NAME);
+        assertEquals(1, results.size());
+        assertEquals(docOneId, results.get(0).getDocId());
+
+        //retrieving document by id
+        final Map<String, Object> knnDocMap = getKnnDoc(INDEX_NAME, docOneId);
+        final Float[] vectorInDocument = getVectorFromDocument(knnDocMap, FIELD_NAME);
+        assertEquals(vectorForDocumentOne.length, vectorInDocument.length);
+        assertArrayEquals(vectorForDocumentOne, vectorInDocument);
```
Comment (jmazanec15) [inline src/test/java/org/opensearch/knn/index/OpenSearchIT.java:432]: Whats the purpose of this assertion? I think the test can remove this to simplify the test.

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java [hunk 3]
```diff
@@ -399,4 +399,71 @@ public void testExistsQuery() throws IOException {
...
+        final Float[] vectorForDocumentOne = {6.0f, 7.0f, 8.0f, 9.0f};
+        final String docOneId = "1";
+        addKnnDoc(INDEX_NAME, docOneId, FIELD_NAME, vectorForDocumentOne);
+
+        final Float[] vectorForDocumentTwo = {2.0f, 1.0f, 3.8f, 2.5f};
+        final String docTwoId = "2";
+        addKnnDoc(INDEX_NAME, docTwoId, FIELD_NAME, vectorForDocumentTwo);
+
+        //checking that one of two documents in retrievable based on closer similarity to query vector
+        int k = 1;
```
Comment (jmazanec15) [inline src/test/java/org/opensearch/knn/index/OpenSearchIT.java:420]: I think we can just make k = 2 and then:
1. Make sure 2 docs are returned by the k-NN query
2. Update doc 2 to null
3. Assert the value returned from retrieving doc is null
4. Make sure 1 doc is returned by the k-NN query when k = 2 (doc id 1)
5. Update doc 2 back to the vector
6. Make sure retrieving doc 2 gives the correct vector
7. Make sure 2 docs are returned by the k-NN query

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java (hunk 2 above)
Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java:432]: Agree, we can assert documents fields after the update only, this particular assertion is redundant. I'll remove it in next commit

File: src/test/java/org/opensearch/knn/index/OpenSearchIT.java (hunk 3 above)
Comment (martin-gaievski) [reply src/test/java/org/opensearch/knn/index/OpenSearchIT.java]: Discussed offline, changing test to use k equals 2 instead of 1 for knn searches

**Review by jmazanec15 — APPROVED** [review]
LGTM thanks!

**Review by VijayanB — APPROVED** [review]
